### ISSUE TITLE
Agda complains about data types in opaque mutual blocks

### ISSUE THREAD

**Issue by nad** [agda/agda#6960]
Why does Agda complain about the mutual block in the following code?
```agda
open import Agda.Builtin.Equality

opaque mutual

  F : Set₁ → Set₁
  F A = A

  data D : Set where

_ : F Set ≡ Set
_ = refl
```
Agda complains about the definition of `D` inside the mutual block:
```
Only functions and primitives can be marked opaque. This
declaration will be treated as transparent.
when scope checking the declaration
  data D where
```
However, `F` is made opaque, so I don't see the need to complain about `D`. Doesn't it suffice to complain if the mutual block only contains things that cannot be made opaque?

**Comment by plt-amy** [comment]
It doesn't complain about the mutual block... it complains about `D`. That definition will not be opaque, since there are no opaque data types (yet).

**Comment by nad** [comment]
Yes, but is there a need to complain about this? You cannot mark only `F` as opaque, because it is in a mutual block, but if you mark the mutual block as opaque, then Agda complains.

**Comment by jespercockx** [comment]
Moving this to the icebox due to a lack of interest/manpower from the development team.
